nova/virt/vmwareapi/cluster_util.py

```diff
@@ -170,52 +152,36 @@ def delete_vm_group(session, cluster, vm_group):
+
+    We only do this for an admin-defined group
+    """
     cluster_config = session._call_method(
         vutil, "get_object_property", cluster, "configurationEx")
 
     client_factory = session.vim.client.factory
     config_spec = client_factory.create('ns0:ClusterConfigSpecEx')
     config_spec.groupSpec = []
-    config_spec.rulesSpec = []
-    for group_info in group_infos:
-        if not group_info.name.startswith(constants.DRS_PREFIX):
-            # We only do this, if this is an admin-defined group, because
-            # VmGroups are not used by the rules created by Nova.
-            group = _get_vm_group(cluster_config, group_info)
-
-            if not group:
-                # Creating group
-                operation = "add"
-            else:
-                # VM group exists on the cluster which is assumed to be
-                # created by VC admin. Add instance to this vm group and let
-                # the placement policy defined by the VC admin take over
-                operation = "edit"
-            group_spec = _create_vm_group_spec(
-                client_factory, group_info, [vm_ref], operation=operation,
-                group=group)
-            config_spec.groupSpec.append(group_spec)
-
-        # If server group policies are defined (by tenants), then
-        # create/edit affinity/anti-affinity rules on cluster.
-        # Note that this might be add-on to the existing vm group
-        # (mentioned above) policy defined by VC admin i.e if VC admin has
-        # restricted placement of VMs to a specific group of hosts, then
-        # the server group policy from nova might further restrict to
-        # individual hosts on a cluster
-        if group_info.policies:
-            # VM group does not exist on cluster
-            policy = group_info.policies[0]
-            if policy != 'soft-affinity':
-                rule_name = "%s-%s" % (group_info.name, policy)
-                rule = _get_rule(cluster_config, rule_name)
-                operation = "edit" if rule else "add"
-                rules_spec = _create_cluster_rules_spec(
-                    client_factory, rule_name, [vm_ref], policy=policy,
-                    operation=operation, rule=rule)
-                config_spec.rulesSpec.append(rules_spec)
+
+    group = _get_vm_group(cluster_config, vm_group_name)
+
+    if not group:
+        # Creating group
+        operation = "add"
+        group = create_vm_group(client_factory, vm_group_name, [vm_ref])
+    else:
+        # VM group exists on the cluster which is assumed to be
+        # created by VC admin. Add instance to this vm group and let
+        # the placement policy defined by the VC admin take over
+        operation = "edit"
+        if not hasattr(group, "vm"):
+            group.vm = [vm_ref]
+        else:
+            group.vm += vm_ref
+
+    group_spec = create_group_spec(client_factory, group, operation)
+    config_spec.groupSpec.append(group_spec)
```

Comment:
I think we might as well move lines 104 + 105 down here and just create groupSpec here, i.e.
```
    config_spec = client_factory.create('ns0:ClusterConfigSpecEx')
    config_spec.groupSpec = [group_spec]
```
We don't use it in between, so I don't see a point in putting them that far from each other.

Reply:
I think you changed that in another function (delete_vm_group). I still think we could change it here, too. My line-number might have been off. I meant 165 and 166 of the new file should move down here.

Reply:
Sorry, now I should have the right line. Was a bit too quick about that.